src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -817,11 +820,14 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
                                           * normalBarotropicVelocitySubcycleCur(eoe) * fEdge(eoe)
                           end do
 
+                          sshCell1 = sshSubcycleCur(cell1) - tidalPotentialEta(cell1)
+                          sshCell2 = sshSubcycleCur(cell2) - tidalPotentialEta(cell2)
+
```

Comment:
Let me think about this. There is a danger in adding extra computation and (even more so) extra memory access for special types of simulations, that are then used by all simulations, even if they compute with zeros. The barotropic subcycling does not scale well and takes a large fraction of the simulation time. These lines are being added to the barotropic subcycling.

Alternatives are to make a separate loop for including tides, or using CPP pragmas. Both come with their own issues of code upkeep and readability.

Reply:
I suppose one option would be to have a ``sshScratch`` variable and do something like this:

```
if (config_use_tidal_potential_forcing) then
  do iCell = 1,nCell
     sshScratch(iCell) = sshSubcycleCur(iCell) - tidalPotentialEta(iCell)
  end do
else
  do iCell = 1,nCell
     sshScratch(iCell) = sshSubcycleCur(iCell)
  end do
end if
.
.
.
do iEdge = 1, nEdges

  cell1 = cellsOnEdge(1,iEdge)
  cell2 = cellsOnEdge(2,iEdge)
 
  normalBarotropicVelocitySubcycleNew(iEdge) =  ...
   * (sshScratch(cell2) - sshScratch(cell1) ) / dcEdge(iEdge) + ...

end do

```
That at least gets rid of the extra indirect memory accesses inside the edge loop, but it still involves another array.

Reply:
This would avoid the extra copy...

```
if (config_use_tidal_potential_forcing) then
  call mpas_pool_get_array(statePool, 'sshSubcycle', sshSubcycleCur, oldBtrSubcycleTime)
  call mpas_pool_get_array(scratchPool, 'sshScratch, sshScratch)
  call mpas_pool_get_array(forcingPool, 'tidalPotentialEta', tidalPotentialEta)
  do iCell = 1,nCell
     sshScratch(iCell) = sshSubcycleCur(iCell) - tidalPotentialEta(iCell)
  end do
else
  call mpas_pool_get_array(statePool, 'sshSubcycle', sshScratch, oldBtrSubcycleTime)
end if
```

Reply:
@mark-petersen, I think this is resolved now based on what we discussed. Let me know if you have other thoughts.